drivers/sensors/lsm6dso32_uorb.c

```diff
@@ -0,0 +1,1979 @@
+ *   priv    - The instance of the LSM6DSO32 sensor.
+ *   addr    - The register address to read from.
+ *   buf     - The buffer of data to read into.
+ *   nbytes  - The number of bytes to read into the buffer.
+ *
+ * Returned Value:
+ *   Zero (OK) on success; a negated errno value on failure.
+ *
+ ****************************************************************************/
+
+static int lsm6dso32_read_bytes(FAR struct lsm6dso32_dev_s *priv,
+                                uint8_t addr, void *buf, size_t nbytes)
+{
+  struct i2c_msg_s cmd[2];
+
+  /* Register addressing part of command. */
+
+  cmd[0].frequency = CONFIG_SENSORS_LSM6DSO32_I2C_FREQUENCY;
+  cmd[0].addr = priv->addr;
+  cmd[0].flags = I2C_M_NOSTOP;
+  cmd[0].buffer = &addr;
+  cmd[0].length = sizeof(addr);
+
+  /* Read data into buffer. */
+
+  cmd[1].frequency = CONFIG_SENSORS_LSM6DSO32_I2C_FREQUENCY;
+  cmd[1].addr = priv->addr;
+  cmd[1].flags = I2C_M_NOSTART | I2C_M_READ;
+  cmd[1].buffer = buf;
+  cmd[1].length = nbytes;
+
+  /* Send command over the wire */
+
+  return I2C_TRANSFER(priv->i2c, cmd, 2);
+}
+
+/****************************************************************************
+ * Name: lsm6dso32_set_bits
+ *
+ * Description:
+ *   Read current value of desired register and change specified bits
+ *   while preserving previous ones.
+ *   NOTE: Clear operation performed before set operation.
+ *
+ * Input Parameters:
+ *   priv    - The instance of the LSM6DSO32 sensor.
+ *   addr    - The register address being changed.
+ *   set_bits - A mask of the bits to be set to 1.
+ *   clear_bits  - A mask of the bits to be set to 0.
+ *
+ * Returned Value:
+ *   Zero (OK) on success; a negated errno value on failure.
+ *
+ ****************************************************************************/
+
+static int lsm6dso32_set_bits(FAR struct lsm6dso32_dev_s *priv, uint8_t addr,
+                              uint8_t set_bits, uint8_t clear_bits)
+{
+  int err;
+  uint8_t reg;
+
+  err = lsm6dso32_read_bytes(priv, addr, &reg, sizeof(reg));
+
+  if (err < 0)
+    {
+      return err;
+    }
+
+  reg = (reg & ~clear_bits) | set_bits;
+  return lsm6dso32_write_bytes(priv, addr, &reg, sizeof(reg));
+}
+
+/****************************************************************************
+ * Name: accel_set_odr
+ *
+ * Description:
+ *      Sets the accelerometer ODR.
+ ****************************************************************************/
+
+static int accel_set_odr(FAR struct lsm6dso32_dev_s *dev,
+                         enum lsm6dso32_odr_e odr)
+{
+  int err;
+
+  err = lsm6dso32_set_bits(dev, CTRL1_XL, (odr & 0xf) << 4, 0xf0);
+
+  if (err < 0)
+    {
+      return err;
+    }
+
+  dev->accel.odr = odr;
+  return err;
+}
+
+/****************************************************************************
+ * Name: gyro_set_odr
+ *
+ * Description:
+ *      Sets the gyroscope ODR.
+ ****************************************************************************/
```

Comment:
the end of function headers should be preceded by a new line, here and in all the other places. Check this issue https://github.com/apache/nuttx/issues/12133